packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Program.cs

```diff
@@ -40,18 +41,38 @@ public static void Main(string[] args)
                 MetricsFilePath = Path.Combine(metricsFolder, $"portingAssistant-telemetry-{DateTime.Today.ToString("yyyyMMdd")}.metrics"),
                 Suffix = new List<string>() { ".log", ".metrics" }
             };
-            var lastReadTokenFile = Path.Combine(teleConfig.LogsPath, "lastToken.json");
-            string prefix = portingAssistantPortingConfiguration.PortingAssistantMetrics["Prefix"].ToString();
-            
+
+            var outputTemplate = "[{Timestamp:yyyy-MM-dd HH:mm:ss} {Level:u3}] {SourceContext}: {Message:lj}{NewLine}{Exception}";
+            var logConfiguration = new LoggerConfiguration().Enrich.FromLogContext()
+                .MinimumLevel.Debug()
+                .WriteTo.RollingFile(
+                    Path.Combine(args[2], "logs", "portingAssistant-telemetry-{Date}.log"),
+                    outputTemplate: outputTemplate,
+                    fileSizeLimitBytes: 1000000);
+
+            Log.Logger = logConfiguration.CreateLogger();
+
             var logTimer = new System.Timers.Timer();
             logTimer.Interval = Convert.ToDouble(portingAssistantPortingConfiguration.PortingAssistantMetrics["LogTimerInterval"].ToString());
 
-            logTimer.Elapsed += (source, e) => LogUploadUtils.OnTimedEvent(source, e, teleConfig, lastReadTokenFile, profile, useDefaultCreds, prefix, shareMetric);
-
+            LogUploadUtils.InitializeUploader(
+                shareMetric,
+                teleConfig,
+                profile,
+                useDefaultCreds,
+                Log.Logger);
+            logTimer.Elapsed += LogUploadUtils.OnTimedEvent;
             logTimer.AutoReset = true;
-
             logTimer.Enabled = true;
 
+            var writeLogErrorsTimer = new System.Timers.Timer
+            {
+                Interval = 300000,
```

Comment:
There is a config in IDE regarding the interval value. Is there any similar config that we can use instead hard-coded for standalone tool?

Reply:
We can add a config value, but I purposely didn't want to add another config field since this only will control how often we write our log upload errors to the log file, so it might be confusing for the customer to want to configure this.